### PR TITLE
New Cantonese Cheat Sheet

### PR DESCRIPTION
**Pull request by wongalvis**
**What does your Instant Answer do?**
Provide a list of basic Cantonese words and phrases

**What problem does your Instant Answer solve (Why is it better than organic links)?**
Instantly displays a list of commonly used and basic Cantonese words and phrases

**What is the data source for your Instant Answer? (Provide a link if possible)**
http://www.cantonese.sheik.co.uk/

**Why did you choose this data source?**
It has plenty of information regarding the words and phrases

**Are there any other alternative (better) data sources?**
Yes, there should be a number of alternative data sources.

**What are some example queries that trigger this Instant Answer?**
"cantonese cheat sheet", "cantonese words reference", "cantonese phrases help"

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
People who want to learn/get started with Cantonese.

**Which existing Instant Answers will this one supercede/overlap with?**
Should be no.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
2015-09-12 14_44_30-cantonese help at duckduckgo

## Checklist
Please place an 'X' where appropriate.

```
- [] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        - [X] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 10

    - Windows 7
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 8
        - [] IE 9
        - [] IE 10

    - Windows XP
        - [] IE 7
        - [] IE 8

    - Mac OSX
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] Safari

    - iOS (iPhone)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - iOS (iPad)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - Android
        - [] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera

```

-----
IA Page: https://duck.co/ia/view/cantonese_cheat_sheet

### CODE REVIEW

**Comment by moollaza** [comment]
@wongalvis thanks a lot! I think seeing as we already have about 3 or 4 language-based cheat sheets that use linguanaut as the sorce, and therefor include the same phrases, we should follow suit here and list out the phrases from http://www.linguanaut.com/english_cantonese.htm

**Comment by moollaza** [comment]
I think we should definitely standardize the section that we want because I can see the numbers and days of the week being useful. Obviously we can't cover too much because the cheat sheet will become cumbersome, but I will ask the team and our translation managers what they think should be the baseline for these language cheat sheets.

**Comment by wongalvis** [comment]
@moollaza The pronunciation provided at http://www.linguanaut.com/english_cantonese.htm are not really English users-friendly, for instance "Good Morning" is pronounced as "Zou2 san4", the number at the end of each word is not a conventional denotation for language pronunciation internationally, and there is a total of 9 of them in Cantonese (which makes Cantonese one of the hardest languages to learn in the world). The pronunciation from the source I've chosen are composed of English alphabets, I also personally added some phrases and pronunciation to make them more suitable for English language speakers.

**Comment by moollaza** [comment]
@wongalvis sorry I didn't mean that you should use their Cantonese phrase, I just meant that we should take the English phrases they have, and use them in the cheat sheet. I'm assuming you're a native speaker so you can translate them yourself?

I'm hoping we can standardize our language cheat sheets so they all contain the same phrases.

**Comment by wongalvis** [comment]
@moollaza I chose them because they are the basic Cantonese words and phrases which can also be an introduction. I'm afraid they will become too complicated for people to know/understand when it comes to translating those phrases you suggested, but after all it's up to you.

**Comment by moollaza** [comment]
Hey @wongalvis I spoke with the team and we agreed that having a standardized set of phrases is ideal. It's not set in stone and we want the community's opinions on what phrases are best, but I think the phrases that Linguanaut are a great start as they cover basic, conversational phrases that someone might need if they perhaps travelling. I think learning a language is something different and these cheat sheets aren't intended to help you learn a language, so much as figure out how to say a common phrase, quickly.

**Comment by wongalvis** [comment]
@moollaza So I should follow the German Cheat Sheet regarding choices of phrases?

**Comment by moollaza** [comment]
@wongalvis yes please.

**Comment by wongalvis** [comment]
Updated @moollaza

**Comment by moollaza** [comment]
@wongalvis thanks! I'll have the team share this so our Community can review the translations and once the translations are agreed on I'll be able to merge it.

**Comment by moollaza** [comment]
@wongalvis apologies for the delay. It looks like our previous announcement didn't garner any feedback. I'll have the team make another announcement. I'd like to make sure at least someone else is OK with this before pushing it live.

**Comment by wongalvis** [comment]
@moollaza Sure :) Merry Christmas :christmas_tree:

**Comment by tagawa** [comment]
Got some [feedback from Twitter](https://twitter.com/u46esp/status/681013629169184768):
*"How are you" is nei ho ma. I never seen people use that phrase "nei tim ah" for how are you exchange but I can be wrong.*

**Comment by wongalvis** [comment]
@tagawa Yes, nei ho ma should be more appropriate. Updated.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Cantonese cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/cantonese_cheat_sheet_mobile.png)

**Comment by wongalvis** [comment]
Hmm... some encoding problem, probably just due to browser settings?

**Comment by tagawa** [comment]
Thanks for the quick update @wongalvis. Don't worry about the encoding issue, it's just in the automatic screenshot tool.

**Comment by wongalvis** [comment]
@tagawa Great, thx!

File: share/goodie/cheat_sheets/json/cantonese.json [hunk 1]
```diff
@@ -0,0 +1,227 @@
+{
+    "id": "cantonese_cheat_sheet", 
+
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/cantonese.json]: @wongalvis let's remove the newlines and normalize the indentation -- it look like some lines are a little off

**Comment by moollaza** [comment]
@wongalvis this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Comment by moollaza** [comment]
@wongalvis I just realized that you have the transliteration inlined here -- we actually now have a new `language` template, and folder for cheat sheets which allows a new property, `trn` that allows you to provide the transliteration so we can put it on its own line.

Can you please adjust that as well?

I'll merge for now because this is still OK.

Thanks again!

**Comment by moollaza** [comment]
@wongalvis congrats! It's live: https://duckduckgo.com/?q=Cantonese%20cheat%20sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Pull request merged**